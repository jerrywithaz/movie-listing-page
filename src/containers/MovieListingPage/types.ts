import { Cinema, SelectedCinema, Session, Film } from "core/types";

export type MovieListingPageMappedProps = {
    selectedTheater: SelectedCinema,
    theaters: Cinema[],
    films: Film[]
};

export type MovieListingPageMappedDispatch = {
    setSelectedTheater: (cinema: Cinema) => void
};

export type MovieListingPageProps = (
    React.HTMLAttributes<HTMLDivElement> & 
    MovieListingPageMappedProps & 
    MovieListingPageMappedDispatch
);