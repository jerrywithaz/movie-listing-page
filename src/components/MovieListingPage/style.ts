import styled from 'styled-components';

export const MovieListingPage = styled.div`
    width: 100%;
    height: 100%;
    background: ${props => props.theme.palette.gray.primary};
`;