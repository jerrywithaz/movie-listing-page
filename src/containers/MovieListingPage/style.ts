import styled from 'styled-components';
import DefaultHero from 'components/Hero';
import DefaultFindAMovie from 'components/FindAMovie';

export const MovieListingPage = styled.div`
    width: 100%;
    height: 100%;
    background: ${props => props.theme.palette.gray.primary};
    display: flex;
    position: relative;
    align-items: center;
    justify-content: center;
`;

export const Hero = styled(DefaultHero)`
    position: absolute;
    width: 100%;
    top: 0;
    left: 0;
    z-index: 1;
`;

export const FindAMovie = styled(DefaultFindAMovie)`
    position: relative;
    z-index: 2;
`;