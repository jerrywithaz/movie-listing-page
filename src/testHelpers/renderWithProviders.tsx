import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import {Provider} from 'react-redux';
import ThemeProvider from 'containers/ThemeProvider';
import configureStore from 'core/store';

type Options = {
  enzyme: boolean,
  fullMount: boolean
};

const initialState = {
  data: {
    market: null,
    selectedCinema: null
  },
  error: null
};

function renderWithProviders(component: JSX.Element, options? : Options) {

    const { enzyme, fullMount } = options || { enzyme: false, fullMount: true };
    const store = configureStore(initialState);
    const node = (
      <Provider store={store}>
        <ThemeProvider>
          {component}
        </ThemeProvider>
      </Provider>
    );

    if (enzyme) {
      if (fullMount) {
        return mount(node);
      }
      else {
        return shallow(node);
      }
    }
    else {
        return renderer.create(node);
    }
    
}

export default renderWithProviders;