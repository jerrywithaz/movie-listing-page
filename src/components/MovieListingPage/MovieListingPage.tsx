import React from 'react';

import * as Styled from './style';

function MovieListingPage() {
    return (
        <Styled.MovieListingPage>

        </Styled.MovieListingPage>
    );
}

export default MovieListingPage;