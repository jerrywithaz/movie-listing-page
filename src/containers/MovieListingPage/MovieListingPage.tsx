import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { Dispatch } from 'redux';
import { makeSelectCinemas, makeSelectSelectedCinema } from 'core/selectors';
import { MovieListingPageProps, MovieListingPageMappedProps, MovieListingPageMappedDispatch } from './types';
import { AppState, Cinema } from 'core/types';
import { setSelectedCinema } from 'core/actions';

import * as Styled from './style';


function MovieListingPage({
    theaters, 
    selectedTheater,
    setSelectedTheater,
    ...rest
}: MovieListingPageProps) {

    return (
        <Styled.MovieListingPage {...rest}>
            <Styled.Hero/>
            <Styled.FindAMovie 
                theaters={theaters} 
                selectedTheater={selectedTheater}
                setSelectedTheater={setSelectedTheater}/>
        </Styled.MovieListingPage>
    );

}

const mapStateToProps = createStructuredSelector<AppState, MovieListingPageMappedProps>({
    theaters: makeSelectCinemas(),
    selectedTheater: makeSelectSelectedCinema()
});

const mapDispatchToProps = (dispatch: Dispatch): MovieListingPageMappedDispatch => {
    return {
        setSelectedTheater: (cinema: Cinema) => dispatch(setSelectedCinema(cinema))
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieListingPage);