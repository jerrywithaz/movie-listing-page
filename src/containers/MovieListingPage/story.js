import React from 'react';
import { storiesOf } from '@storybook/react';
import MovieListingPage from '.';
import { getMockFilms, getMockTheaters } from 'core/mocks';

const story = storiesOf('MovieListingPage', module);

const films = getMockFilms();
const theaters = getMockTheaters();

story.add('Default', () => <MovieListingPage films={films} theaters={theaters}/>);
  