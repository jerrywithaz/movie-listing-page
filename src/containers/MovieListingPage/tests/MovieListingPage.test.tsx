import React from 'react';
import MovieListingPage from '..';
import renderWithProviders from 'testHelpers/renderWithProviders';
import { ReactTestRenderer } from 'react-test-renderer';

describe("containers/MovieListingPage", () => {

    it("should match snapshot", () => {

        const renderer = renderWithProviders(<MovieListingPage/>) as ReactTestRenderer;
        const tree = renderer.toJSON() ;

        expect(tree).toMatchSnapshot();

    });

});