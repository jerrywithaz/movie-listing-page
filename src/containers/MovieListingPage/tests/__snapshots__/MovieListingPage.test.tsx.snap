// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/MovieListingPage should match snapshot 1`] = `
.c1 {
  background: linear-gradient( 159.41deg, #FF009D 3.42%, #F5B324 96.5% );
  height: 560px;
  width: 100%;
  position: relative;
  overflow: hidden;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  position: absolute;
  background: #E5E5E5;
  bottom: -34%;
  content: '';
  display: block;
  height: 50%;
  left: 0;
  right: 0;
  -webkit-transform-origin: 100%;
  -ms-transform-origin: 100%;
  transform-origin: 100%;
  -webkit-transform: skewY(-3deg);
  -ms-transform: skewY(-3deg);
  transform: skewY(-3deg);
  z-index: -1;
}

.c4 {
  position: relative;
}

.c5 {
  width: 260px;
  height: auto;
  display: block;
  margin: 0 auto;
}

.c6 {
  width: 552px;
  height: auto;
  display: block;
  margin: 0 auto;
}

.c11 {
  font-size: 26px;
  line-height: 30px;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  color: #000000;
  text-transform: uppercase;
  margin: 0px 0px 24px 0px;
}

.c14 {
  font-size: 16px;
  line-height: 20px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  color: #6B6B6B;
  text-transform: uppercase;
  margin: 24px 0px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 196px;
  margin-bottom: 32px;
  overflow: hidden;
}

.c15 {
  margin-bottom: 16px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

.c7 {
  height: 722px;
  width: 752px;
  box-shadow: 0px 1px 5px rgba(24,17,3,0.2), 0px 3px 4px rgba(0,0,0,0.12), 0px 2px 4px rgba(24,17,3,0.14);
  border-radius: 4px;
  padding: 32px;
  background: #ffffff;
}

.c12 {
  display: block;
  text-align: center;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 560px;
  margin: 0 auto;
  max-width: 100%;
  height: 100%;
  overflow: hidden;
}

.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  position: relative;
  width: 100%;
  top: 0;
  left: 0;
  z-index: 1;
}

.c9 {
  position: relative;
  z-index: 2;
  margin-top: -150px;
  margin-bottom: 32px;
}

@media (max-width:480px) {
  .c1 {
    height: 460px;
  }
}

@media (max-width:992px) {
  .c6 {
    max-width: calc(100% - 48px);
  }
}

@media (max-width:992px) {
  .c8 {
    max-width: calc(100% - 64px);
  }
}

<div
  className="c0"
>
  <header
    className="c1 c2"
  >
    <div
      className="c3"
    />
    <div
      className="c4"
    >
      <img
        alt="Welcome to"
        className="c5"
        src="hero-arrow.svg"
      />
      <img
        alt="Alamo Drafthouse Cinema"
        className="c6"
        src="hero-logo.png"
      />
    </div>
  </header>
  <main
    aria-live="polite"
    className="c7 c8 c9"
    role="main"
  >
    <div
      className="c10"
    >
      <h1
        className="c11 c12"
      >
        Find a Movie
      </h1>
      <section
        aria-labelledby="heading-select-theater"
        className="c13"
      >
        <h2
          className="c14 c15"
          id="heading-select-theater"
        >
          Select Theater
        </h2>
        <div
          className="c16"
        />
      </section>
    </div>
  </main>
</div>
`;
